frontend/src/lib/Utils.tsx

```diff
@@ -117,6 +117,31 @@ export function getRunDurationFromWorkflow(workflow?: Workflow): string {
   return getDuration(new Date(workflow.status.startedAt), new Date(workflow.status.finishedAt));
 }
 
+/**
+ * Calculate the time duration a task has taken as a node in workflow. If start time or end time
+ * is not available, return '-'.
+ *
+ * @param workflow
+ * @param nodeId
+ */
+export function getRunDurationFromNode(workflow: Workflow, nodeId: string): string {
+  if (
+    !workflow ||
+    !workflow.status ||
+    !workflow.status.nodes ||
+    !workflow.status.nodes[nodeId] ||
+    !workflow.status.nodes[nodeId].startedAt ||
+    !workflow.status.nodes[nodeId].finishedAt
```

Comment:
this can now be simplified as
```
const node=workflow?.status?.nodes[nodeId]
if (!node || !node.startedAt || !node.finishedAt) {
```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Many of KFP UI code was written before the feature exists, so you may see existing code that are more verbose than necessary.